lib/sql-logging/configuration.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+module SqlLogging
+  class Configuration
+    DEFAULTS = {
+      show_sql_backtrace: true,
+      show_top_sql_queries: :total_time,
+      top_sql_queries: 10,
+      backtrace_cleaner: nil,
+      logger: nil
+    }.freeze
+
+    attr_accessor(*DEFAULTS.keys)
```

Comment:
On a personal note I would have defined 2 methods for each of these since we end up with 3 out of 10 that gets overridden, but I'm also ok keeping it that way.